### ISSUE TITLE
Search by expression (advanced)

### ISSUE THREAD

**Issue by hlesaint** [openmindanesth/openmind#61]
I didn't thought about it earlier, but it would be great to enable the following possibilities in the search by expression window:
– search multiple expressions, e.g.: mice medetomidine electrostimulation 
– search for author/investigator, year of publication/discovery, expressions in abstract, key words
> For instance: author:Nasrallah, year:2014, abstract:"blood pressure", key:fMRI

**Comment by tgetgood** [comment]
I'm with you on that; advanced search is going to be important once there's any quantity of data in there.

The search right now is very basic and needs improvement in any case. I'll look at what options elastic search has for us to use.

**Comment by hlesaint** [comment]
"Search by expression" bar should also search for tags. If I tag an extract with isoflurane I should find it again by just typing "isoflurane" in the search bar. 

If I want to search particularly for the tag isoflurane I should find it by typing "**tag:isoflurane**". This is redundant with clicking the tags and makes life easier for people used to work in the terminal. It would be great if it is possible to complete tag expressions with tab to see what's there.

**Comment by tgetgood** [comment]
Fields that should be included in search

- extract 
- tags
- extract author name
- figure caption? 
- anything else?

**Comment by hlesaint** [comment]
**Basics** included in search (to keep it specific) should probably be
- extract
- tags
- article author name(s) /investigator (for lab notes); here I would go for entire author list, e.g. Nils Brose, "Nils Brose"
- publication /defined investigation year (and year range, e.g., 2017-2020)


Further interesting for an **extended search** would be
- article title /lab note title
- article key words
- article abstract /lab note description
- figure caption

These could come in a package when doing an "extended search", e.g., [ext]:search terms
In this case it may help to have a radio button for _extended search_.


It would be also great to define **search specifications**; ideas /suggestions would be:
- isoflurane **author:"Nils Brose"**; searches for _extract author_ name; no search term: all extracts created by author
- isoflurane **author1:"Nils Brose"**; searches for _extract first author_ name; no search term: all extracts created by 1st author
- isoflurane **doi:10.3389/fnsys.2020.00008**; searches for extracts created from articles with a specific doi; no search term: all extracts of that article
- isoflurane **year:2019**; publication /defined investigation year 
- isoflurane **year:2017-2020**; range of publication /defined investigation year 
- isoflurane **extauthor:"Nils Brose"**;  search for _extract author /creator_ would be also interesting
- isoflurane **extyear:2017-2020**; maybe also interesting
- **tag:isoflurane**; redundant with tag

**Specifications** for logical AND/OR would be also great, for ONE/ALL OF THE TERMS.

I think only the **basics** and **AND/OR** (or most importantly search for multiple expressions, e.g.: mice medetomidine electrostimulation) are necessary for now. The more advanced steps can be developed (and discussed) after ISMRM.

**Comment by tgetgood** [comment]
We need to distinguish somehow between the author of the extract and the authors of the source paper. 

I've broadened the search so that the extract author, doi, and tags are also searched when you type. I think it already makes the search a lot more productive.

The next step now is to implement the filtering DSL.

Would an advanced search page, like one you would find in a library catalogue, be appropriate? It's pretty heavy duty, but at least it gives you a way to interact with the search system. If we just have a secret language for searching by, say, `doi`, very few people will know it. This becomes even more problematic as we invent clever shorthands like `author1` or `extyear`.

**Comment by hlesaint** [comment]
That's great. The extended search makes a huge difference. 

Currently, multiple search terms seem to broaden the search instead of making it more specific, e.g., 'medetomidine mice' gives you all extracts with medetomidine and all extracts with mice. This makes sense as an advanced search option, but as a default for combined search terms it might be beneficial to narrow down your search to ONLY extracts that include both mice and medetomidine. 

I think for the standard search I'd go with the article's author name /investigator (for lab notes) as this is what you are usually interested in, particularly first and last author (refers to the lab). The extract creator might become increasingly interesting when you are more familiar with the content on OpenMind and should be searchable on demand. 

Having terms like **tag:isoflurane** makes only sense when the information is at hand. One idea would be to have a symbol at the search bar, which expands a legend on click: 

![2](https://user-images.githubusercontent.com/44240982/88345797-0f393180-cd47-11ea-9336-b46c17a4e9b8.png)

A convenient way would be to click a desired expressions and they are automatically filled into the search bar so you only need to continue with writing the search term:

![Ohne Titel](https://user-images.githubusercontent.com/44240982/88345846-368ffe80-cd47-11ea-9bc2-59efff50aedc.png)

This is fast when you don't know the expression and good to learn, so you soon know your key words, which you then just write into the input box to specify your search.

An advanced search page could be something for a later stage.

**Comment by hlesaint** [comment]
I think I would also include the **exact publication year** of an article (and maybe investigation year for lab notes) as part of the default search. If you want to search for a key finding of a specific paper you probably wanna type: _isoflurane Suzuki 2020_, because this is what you mostly remember.

**Comment by hlesaint** [comment]
Multiple search terms seem to broaden the search instead of making it more specific, e.g., 'medetomidine mice' gives you all extracts with medetomidine and all extracts with mice. This makes sense as an advanced search option, but as a default for combined search terms it might be beneficial to narrow down your search to ONLY extracts that include both mice and medetomidine (as this is the case when searching via tags).

**Comment by tgetgood** [comment]
Yes, right now it is an OR of all terms. I'll change that as part of advanced search.
